incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1983,26 +1946,23 @@ private void onDecodeSubscribe(
                 if (session)
                 {
                     final MqttSessionStateFW.Builder sessionStateBuilder =
-                        mqttSessionStateFW.wrap(sessionStateBuffer, 0, sessionStateBuffer.capacity())
-                            .clientId(clientId);
+                        mqttSessionStateFW.wrap(sessionStateBuffer, 0, sessionStateBuffer.capacity());
 
-                    newSubscriptions.forEach(subscription ->
+                    List<Subscription> currentSubscriptions = new ArrayList<>(sessionStream.subscriptions);
+                    currentSubscriptions.addAll(newSubscriptions);
+                    sessionStream.unAckedSubscriptions.addAll(newSubscriptions);
+                    currentSubscriptions.forEach(subscription ->
+                        sessionStateBuilder.subscriptionsItem(subscriptionBuilder ->
                         {
-                            sessionStateBuilder.subscriptionsItem(subscriptionBuilder ->
-                            {
-                                subscriptionBuilder.subscriptionId(subscription.id);
-                                subscriptionBuilder.flags(subscription.flags);
-                                subscriptionBuilder.pattern(subscription.filter);
-                            });
-                            sessionStream.unAckedSubscriptions.add(subscription);
-                        }
+                            subscriptionBuilder.subscriptionId(subscription.id);
+                            subscriptionBuilder.flags(subscription.flags);
+                            subscriptionBuilder.pattern(subscription.filter);
```

Comment:
Use fluent API, chaining these into a single statement, eliminate need for `{...}` braces and reduce size of lambda variable, say `sub` and `state`.

Do we need to make a copy of the combined collection to build the session state?
Perhaps we could just iterate `sessionStream.subscriptions` and then `newSubscriptions`?